### DISCUSSION TITLE
Uploading Docker images and updating existing Service

### DISCUSSION THREAD

**Discussion by jonasgroendahl** [pulumi/pulumi#10525] [Pulumi in Practice]
Hello all!

So i was thinking to create a seperate stack in my Pulumi project that simply uploads a Docker image to ECR, create new task definition and then reference an existing service to update the task definition.

Is this at at possible?

I'm thinking something in the line of this:

```
const image = new docker.Image()

const taskDefinition = new aws.ecs.TaskDefinition(prefix(`TaskDefinition`)

const service = await aws.ecs.getService({
    clusterArn: "x",
    serviceName: "y"
  })

service.taskDefition = taskDefinition

```

But perhaps I got the thinking wrong, I should probably export the task definition in this seperate stack and use it in the main stack where the service is initialized?

**Comment by nimbinatus** [comment]
Is there a reason you need to create a new stack for this? I'm not quite sure why you need a new stack; think of stacks more like different environments than separate workflows. You can use the [dependsOn](https://www.pulumi.com/docs/intro/concepts/resources/options/dependson/) option to cause the other resources in the main stack to wait until the service is initialized.
